pkg/kubectl/cmd/cmd_test.go

```diff
@@ -219,167 +228,333 @@ func NewTestFactory() (*cmdutil.Factory, *testFactory, runtime.Codec, runtime.Ne
+	return f.tf.Printer, f.tf.Err
+}
+
+func (f *fakeFactory) Scaler(*meta.RESTMapping) (kubectl.Scaler, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) Reaper(*meta.RESTMapping) (kubectl.Reaper, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) HistoryViewer(*meta.RESTMapping) (kubectl.HistoryViewer, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) Rollbacker(*meta.RESTMapping) (kubectl.Rollbacker, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) StatusViewer(*meta.RESTMapping) (kubectl.StatusViewer, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) MapBasedSelectorForObject(runtime.Object) (string, error) {
+	return "", nil
+}
+
+func (f *fakeFactory) PortsForObject(runtime.Object) ([]string, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) ProtocolsForObject(runtime.Object) (map[string]string, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) LabelsForObject(runtime.Object) (map[string]string, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) LogsForObject(object, options runtime.Object) (*restclient.Request, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) PauseObject(runtime.Object) (bool, error) {
+	return false, nil
+}
+
+func (f *fakeFactory) ResumeObject(runtime.Object) (bool, error) {
+	return false, nil
+}
+
+func (f *fakeFactory) Validator(validate bool, cacheDir string) (validation.Schema, error) {
+	return f.tf.Validator, f.tf.Err
+}
+
+func (f *fakeFactory) SwaggerSchema(unversioned.GroupVersionKind) (*swagger.ApiDeclaration, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) DefaultNamespace() (string, bool, error) {
+	return f.tf.Namespace, false, f.tf.Err
+}
+
+func (f *fakeFactory) Generators(string) map[string]kubectl.Generator {
+	return nil
+}
+
+func (f *fakeFactory) CanBeExposed(unversioned.GroupKind) error {
+	return nil
+}
+
+func (f *fakeFactory) CanBeAutoscaled(unversioned.GroupKind) error {
+	return nil
+}
+
+func (f *fakeFactory) AttachablePodForObject(runtime.Object) (*api.Pod, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) UpdatePodSpecForObject(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error) {
+	return false, nil
+}
+
+func (f *fakeFactory) EditorEnvs() []string {
+	return nil
+}
+
+func (f *fakeFactory) PrintObjectSpecificMessage(obj runtime.Object, out io.Writer) {
+}
+
+func (f *fakeFactory) Command() string {
+	return ""
+}
+
+func (f *fakeFactory) BindFlags(flags *pflag.FlagSet) {
+}
+
+func (f *fakeFactory) BindExternalFlags(flags *pflag.FlagSet) {
+}
+
+func (f *fakeFactory) PrintObject(cmd *cobra.Command, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error {
+	return nil
+}
+
+func (f *fakeFactory) PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool) (kubectl.ResourcePrinter, error) {
+	return nil, nil
+}
+
+func (f *fakeFactory) NewBuilder() *resource.Builder {
+	return nil
```

Comment:
The `DrainOptions.SetupDrain` in `cmd/drain.go` will use `NewBuidler` of factory, and the default implementation of `NewFactory` will take `Factory::Object`'s return value as parameters. That's what I mean in slack.

But we're luck, it seems `DrainOptions.SetupDrain` is not used by anyone :).